### PR TITLE
Column subset

### PR DESCRIPTION
**Pull request by diegoquintanav**
Hi, would you check the changes I did to the `Schema` definition? It should allow to validate dataframes against subsets of the Schema. You can check what it does in the `./examples/example.py`, which I changed accordingly. I also wrote some basic tests. 

Let me know if this is working in your environment and if it you find it an improvement.

best regards!

### CODE REVIEW

File: pandas_schema/schema.py [hunk 1]
```diff
@@ -38,14 +42,28 @@ def validate(self, df: pd.DataFrame) -> typing.List[ValidationWarning]:
...
-        # It's an error if the number of columns in the schema and data frame are different
         df_cols = len(df.columns)
-        schema_cols = len(self.columns)
-        if df_cols != schema_cols:
-            errors.append(ValidationWarning(
-                'Invalid number of columns. The schema specifies {}, but the data frame has {}'.format(schema_cols,
+
+        # If no columns are passed, validate against every column in the schema
+        # This is the default behaviour
+        if columns == []:
```
Comment (TMiguelT) [inline pandas_schema/schema.py]: columns should default to `None`, and you should check `is None` here instead. Why allocate an empty array when it won't be used for anything?

File: test/test_schema.py [hunk 2]
```diff
@@ -13,36 +13,44 @@ class UnorderedSchema(unittest.TestCase):
     ], ordered=False)
 
     def test_fields(self):
-        self.assertEqual(len(self.schema.columns), 2, 'The schema is not storing all of its columns')
-        self.assertEqual(self.schema.ordered, False, 'The schema is not storing the correct value of ordered')
+        self.assertEqual(len(self.schema.columns), 2,
```
Comment (TMiguelT) [inline test/test_schema.py]: Don't do this either

File: test/test_schema.py [hunk 3]
```diff
@@ -58,6 +66,81 @@ def test_mixed_columns(self):
         self.assertEqual(results[0].column, 'b',
                          'The Schema object is not associating columns and column schemas by name')
 
+    def test_column_subset_detect(self):
```
Comment (TMiguelT) [inline test/test_schema.py]: Good tests

File: pandas_schema/schema.py [hunk 4]
```diff
@@ -10,6 +10,7 @@ class Schema:
     """
     A schema that defines the columns required in the target DataFrame
     """
+    print("Look rick I'm a schema1")
```
[the remark this answers is not in the document]
Comment (diegoquintanav) [reply pandas_schema/schema.py]: I'll delete that. It was a lame debug to prove something to someone.

File: pandas_schema/schema.py [hunk 5]
```diff
@@ -18,18 +19,21 @@ def __init__(self, columns: typing.Iterable[Column], ordered: bool = False):
             the header names. False if the columns should be associated by column header names only. Defaults to False
         """
         if not columns:
-            raise PanSchInvalidSchemaError('An instance of the schema class must have a columns list')
+            raise PanSchInvalidSchemaError(
```
[the remark this answers is not in the document]
Comment (diegoquintanav) [reply pandas_schema/schema.py]: Sure.

**Comment by TMiguelT** [comment]
Could you please remove all the line breaks you've added. It might be your IDE doing it.

Also, I'd rather you didn't change example.py since it's supposed to be the minimal example of the library. It doesn't need to show your new feature since it's more of an edge case

**Comment by diegoquintanav** [comment]
Yep, it's my IDE autoformatting lines. I think it improves readability in split screens, but I will remove it. I also removed the unnecesary examples.

Thanks for your comments, they are really helpful.

**Comment by diegoquintanav** [comment]
Can you try passing a single column name to validate? I'm having errors :dagger:

File: pandas_schema/schema.py [hunk 6]
```diff
@@ -24,12 +23,13 @@ def __init__(self, columns: typing.Iterable[Column], ordered: bool = False):
             raise PanSchInvalidSchemaError('The columns field must be a list of Column objects')
 
         if not isinstance(ordered, bool):
-            raise PanSchInvalidSchemaError('The ordered field must be a boolean')
+            raise PanSchInvalidSchemaError(
```
Comment (TMiguelT) [inline pandas_schema/schema.py]: This

File: example/example.py [hunk 7]
```diff
@@ -20,4 +20,4 @@
 errors = schema.validate(test_data)
 
 for error in errors:
-    print(error)
+    print(error)
```
Comment (TMiguelT) [inline example/example.py]: This

File: test/test_example.py [hunk 8]
```diff
@@ -30,3 +30,4 @@ def test_boolean(self):
             code = compile(code_file.read(), str(code_path), 'exec')
             exec(code)
             self.assertEqual(stdout.getvalue(), result_file.read())
+
```
Comment (TMiguelT) [inline test/test_example.py]: This

File: example/example.py (hunk 7 above)
Comment (diegoquintanav) [reply example/example.py]: sorry, what?

File: test/test_example.py (hunk 8 above)
Comment (diegoquintanav) [reply test/test_example.py]: sorry again, what do you mean?

Comment (TMiguelT) [reply test/test_example.py]: You removed the linebreak at the end of the file

Comment (TMiguelT) [reply test/test_example.py]: Or rather, you added some spaces after the existing linebreak

File: example/example.py (hunk 7 above)
Comment (TMiguelT) [reply example/example.py]: I believe you removed the linebreak at the end of the file

**Comment by TMiguelT** [comment]
In general, look at the GitHub diff, and if you've changed things that aren't your code (e.g. formatting), then you should try to fix it

**Comment by diegoquintanav** [comment]
I'm not sure If I fixed it. Would you create a gitter chat or reach me on telegram? I'm @DiegoQuintana.

**Comment by diegoquintanav** [comment]
I'm lost with the diffs. Is everything alright? :scream:

**Comment by TMiguelT** [comment]
Ah look, don't worry. I'll pull it and then tweak it a bit myself

**Comment by TMiguelT** [comment]
For your reference, the things I cleaned up are here: https://github.com/TMiguelT/PandasSchema/commit/75b48995ba7d464cfe830337d79c78a0936c9f28

**Comment by diegoquintanav** [comment]
Cool, the static column type was my bad, I thought I had it fixed. About the rest, thanks! Perhaps for the next PR things get better. This is kind of my first contribution ever.

**Pull request merged**